cpp/tests/utilities_tests/lists_column_wrapper_tests.cu

```diff
@@ -0,0 +1,906 @@
+
+    auto child_offsets = childv.offsets();
+    EXPECT_EQ(child_offsets.size(), 7);
+    test::fixed_width_column_wrapper<size_type> e_child_offsets({0, 2, 2, 5, 5, 6, 8});
+    test::expect_columns_equal(e_child_offsets, child_offsets);
+
+    auto child_data = childv.child();
+    EXPECT_EQ(child_data.size(), 8);
+    test::fixed_width_column_wrapper<T> e_child_data({1, 2, 5, 6, 7, 8, 9, 10});
+    test::expect_columns_equal(e_child_data, child_data);
+  }
+}
+
+TYPED_TEST(ListColumnWrapperTestTyped, ListOfListOfListsWithValidity)
+{
+  using namespace cudf;
+  using T = TypeParam;
+
+  auto valids = cudf::test::make_counting_transform_iterator(
+    0, [](auto i) { return i % 2 == 0 ? true : false; });
+
+  // List<List<List<T>>>, 2 rows
+  //
+  // List<List<List<T>>>:
+  // Length : 2
+  // Offsets : 0, 2, 4
+  // Children :
+  //    List<List<T>>:
+  //    Length : 4
+  //    Offsets : 0, 2, 2, 4, 6
+  //    Null count: 1
+  //    1101
+  //    Children :
+  //      List<T>:
+  //      Length : 6
+  //      Offsets : 0, 2, 4, 6, 8, 11, 12
+  //      Children :
+  //        1, 2, 3, 4, 10, 20, 30, 40, 50, 60, 70, 0
+  {
+    // equivalent to  { {{{1, 2}, {3, 4}}, NULL}, {{{10, 20}, {30, 40}}, {{50, 60, 70}, {0}}} }
+    test::lists_column_wrapper<T> list{{{{{1, 2}, {3, 4}}, {{5, 6, 7}, {0}}}, valids},
+                                       {{{10, 20}, {30, 40}}, {{50, 60, 70}, {0}}}};
+
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 2);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 3);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2, 4});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto child = lcv.child();
+    lists_column_view childv(child);
+    EXPECT_EQ(childv.size(), 4);
+    EXPECT_EQ(childv.null_count(), 1);
+
+    auto child_offsets = childv.offsets();
+    EXPECT_EQ(child_offsets.size(), 5);
+    test::fixed_width_column_wrapper<size_type> e_child_offsets({0, 2, 2, 4, 6});
+    test::expect_columns_equal(e_child_offsets, child_offsets);
+
+    auto child_child = childv.child();
+    lists_column_view child_childv(child_child);
+    EXPECT_EQ(child_childv.size(), 6);
+
+    auto child_child_offsets = child_childv.offsets();
+    EXPECT_EQ(child_child_offsets.size(), 7);
+    test::fixed_width_column_wrapper<size_type> e_child_child_offsets({0, 2, 4, 6, 8, 11, 12});
+    test::expect_columns_equal(e_child_child_offsets, child_child_offsets);
+
+    auto child_child_data = child_childv.child();
+    EXPECT_EQ(child_child_data.size(), 12);
+    test::fixed_width_column_wrapper<T> e_child_child_data(
+      {1, 2, 3, 4, 10, 20, 30, 40, 50, 60, 70, 0});
+    test::expect_columns_equal(e_child_child_data, child_child_data);
+  }
+}
+
+TYPED_TEST(ListColumnWrapperTestTyped, EmptyLists)
+{
+  using namespace cudf;
+  using T = TypeParam;
+
+  // to disambiguiate between {} == 0 and {} == List{0}
+  using L = test::lists_column_wrapper<T>;
```

Comment:
Could this be "LIST" or "List" perhaps? The "L" was throwing me.
Back in my unicode days you could make a unicode string using `L"abc"` 
Also, 'L' can be used to identify long literals values like `2L` so I thought this was make long integers.

Reply:
Right.  The thinking was "keep it short" to try and avoid cluttering the already eyeball-bending pile of brackets.

Reply:
Dave suggested an opinion from @codereport and that maybe
`LCW{}`
is a good clarification

Reply:
Trying it out, it looks good to me.  Actually maybe even better than L, since the L can get lost:

`{{{L{}}}}`
vs
`{{{LCW{}}}}`